src/utils/updates.ts

```diff
@@ -15,72 +15,104 @@ import { PursesJSONState } from '@agoric/wallet-backend';
 const watchGovernance = async (
   leader: Leader,
   unserializer: Marshal<any>,
-  setGovernedParams: (params: GovernedParams) => void
+  setGovernedParamsIndex: ContractSetters['setGovernedParamsIndex'],
+  anchorPetname: string
 ) => {
-  const f = makeFollower(dappConfig.GOVERNANCE_KEY, leader, { unserializer });
+  // E.g. ':published.psm.IST.AUSD.governance'
+  const spec = dappConfig.INSTANCE_PREFIX + anchorPetname + '.governance';
+  const f = makeFollower(spec, leader, { unserializer });
 
   for await (const { value } of iterateLatest(f)) {
     const current = value.current;
     const GiveStableFee = current.GiveMintedFee.value;
     const MintLimit = current.MintLimit.value;
     const WantStableFee = current.WantMintedFee.value;
 
-    setGovernedParams({ GiveStableFee, MintLimit, WantStableFee });
+    setGovernedParamsIndex([
+      [anchorPetname, { GiveStableFee, MintLimit, WantStableFee }],
+    ]);
   }
 };
 
 const watchMetrics = async (
   leader: Leader,
   unserializer: Marshal<any>,
-  setMetrics: (metrics: Metrics) => void
+  setMetricsIndex: ContractSetters['setMetricsIndex'],
+  anchorPetname: string
 ) => {
-  const f = makeFollower(dappConfig.METRICS_KEY, leader, { unserializer });
+  // E.g. ':published.psm.IST.AUSD.metrics'
+  const spec = dappConfig.INSTANCE_PREFIX + anchorPetname + '.metrics';
+  const f = makeFollower(spec, leader, { unserializer });
 
   for await (const { value } of iterateLatest(f)) {
-    setMetrics(value);
+    setMetricsIndex([[anchorPetname, value]]);
   }
 };
 
-const loadInstanceId = async (
+const watchInstanceIds = async (
   leader: Leader,
-  setInstanceId: (id: string | undefined) => void
+  setters: ContractSetters,
+  walletUnserializer: Marshal<any>
 ) => {
   const f = makeFollower(dappConfig.INSTANCES_KEY, leader, {
     unserializer: identityMarshal,
     proof: 'none',
   });
 
+  const watchedAnchors = new Set();
+
   for await (const { value } of iterateLatest(f)) {
-    const mappedEntries = new Map<string, string>(value);
-    setInstanceId(mappedEntries.get('psm'));
+    // Convert entries of ["psm-IST-AUSD", "board012"] to ["AUSD", "board012"].
+    const PSMEntries = (value as [string, string][])
```

Comment:
more concisely this is removing the "psm-IST" prefix.

Let's be clear about the scope of the resulting entries

also no need to capitalize PSM. initial capital letter is usually reserved for types (and classes, which define types)

```suggestion
    // Remove "psm-IST-" prefix so they're like ["AUSD", "board012"]
    const psmIstEntries = (value as [string, string][])
```